lib/components/ModalPortal.js

```diff
@@ -18,194 +18,218 @@ var CLASS_NAMES = {
     };
-  },
+    this.shouldClose = null;
+  }
 
-  componentDidMount: function() {
+  componentDidMount () {
     // Focus needs to be set when mounting and already open
     if (this.props.isOpen) {
       this.setFocusAfterRender(true);
       this.open();
     }
-  },
-
-  componentWillUnmount: function() {
-    clearTimeout(this.closeTimer);
-  },
+  }
 
-  componentWillReceiveProps: function(newProps) {
+  componentWillReceiveProps (newProps) {
     // Focus only needs to be set once when the modal is being opened
     if (!this.props.isOpen && newProps.isOpen) {
       this.setFocusAfterRender(true);
       this.open();
     } else if (this.props.isOpen && !newProps.isOpen) {
       this.close();
     }
-  },
+  }
 
-  componentDidUpdate: function () {
+  componentDidUpdate () {
     if (this.focusAfterRender) {
       this.focusContent();
       this.setFocusAfterRender(false);
     }
-  },
+  }
 
-  setFocusAfterRender: function (focus) {
+  componentWillUnmount () {
+    clearTimeout(this.closeTimer);
+  }
+
+  setFocusAfterRender (focus) {
     this.focusAfterRender = focus;
-  },
+  }
 
-  open: function() {
+  open () {
     if (this.state.afterOpen && this.state.beforeClose) {
       clearTimeout(this.closeTimer);
       this.setState({ beforeClose: false });
     } else {
       focusManager.setupScopedFocus(this.node);
       focusManager.markForFocusLater();
-      this.setState({isOpen: true}, function() {
-        this.setState({afterOpen: true});
+      this.setState({ isOpen: true }, () => {
+        this.setState({ afterOpen: true });
 
         if (this.props.isOpen && this.props.onAfterOpen) {
           this.props.onAfterOpen();
         }
-      }.bind(this));
+      });
     }
-  },
+  }
 
-  close: function() {
-    if (this.props.closeTimeoutMS > 0)
+  close () {
+    if (this.props.closeTimeoutMS > 0) {
       this.closeWithTimeout();
-    else
+    } else {
       this.closeWithoutTimeout();
-  },
+    }
+  }
 
-  focusContent: function() {
+  focusContent () {
     // Don't steal focus from inner elements
     if (!this.contentHasFocus()) {
-      this.refs.content.focus();
+      this.content.focus();
     }
-  },
+  }
 
-  closeWithTimeout: function() {
-    this.setState({beforeClose: true}, function() {
+  closeWithTimeout () {
+    this.setState({ beforeClose: true }, () => {
       this.closeTimer = setTimeout(this.closeWithoutTimeout, this.props.closeTimeoutMS);
-    }.bind(this));
-  },
+    });
+  }
 
-  closeWithoutTimeout: function() {
+  closeWithoutTimeout () {
     this.setState({
       beforeClose: false,
       isOpen: false,
       afterOpen: false,
     }, this.afterClose);
-  },
-
-  afterClose: function() {
-    focusManager.returnFocus();
-    focusManager.teardownScopedFocus();
-  },
+  }
 
-  handleKeyDown: function(event) {
-    if (event.keyCode == 9 /*tab*/) scopeTab(this.refs.content, event);
-    if (event.keyCode == 27 /*esc*/) {
+  handleKeyDown = (event) => {
```

Comment:
Changing it here doesn't work because we want the arrow functions binding.  See section 5 of [this blog post](https://medium.com/@housecor/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56#f45f) for more info.